java/splitter/splitter-core/src/main/java/sleeper/splitter/find/PartitionSplitCheck.java

```diff
@@ -80,7 +80,7 @@ public boolean isNeedsSplitting() {
     }
 
     public boolean maySplitIfCompacted() {
-        return getEstimatedRecordsAfterCompaction() >= splitThreshold;
+        return partition.isLeafPartition() && estimatedRecordsFromReferencesInPartitionTree >= splitThreshold;
```

Comment:
Brackets would be useful for clarity over with are the conditionals to create the boolean
Fine as is though

Reply:
I've split out the check for if it's a leaf partition, thanks.